File: Day033_2025/Q033_Leetcode.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        // Q. Check if Array Is Sorted and Rotated
        int numsSize = nums.size();
        int count = 0;
        for(int numsIndex = 0; numsIndex < nums.size(); ++numsIndex){
            if(nums[numsIndex] > nums[(numsIndex + 1) % numsSize]){
                count += 1;
            }
        }
        return count <= 1;
    }
};

File: Day033_2025/Q033_Leetcode.java
class Q033_Leetcode {
    public boolean check(int[] nums) {
        int count = 0, numsSize = nums.length;

        for (int i = 0; i < numsSize; i++) {
            if (nums[i] > nums[(i + 1) % numsSize]) {
                count++;
            }
        }
        return count <= 1;
    }
}
